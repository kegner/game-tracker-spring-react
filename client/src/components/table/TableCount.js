import React from 'react'
import { useSelector } from 'react-redux';

const TableCount = (props) => {

  const { pageIndex, itemsPerPage } = props;

  const { games } = useSelector(state => state.gameReducer);

  return (
    <>
      {"Showing " + ((pageIndex * itemsPerPage) + (games.length > 0 ? 1 : 0)) +
      " to " + Math.min(games.length, (pageIndex + 1) * itemsPerPage) +
      " of " + games.length}
    </>
  )
}

export default TableCount
