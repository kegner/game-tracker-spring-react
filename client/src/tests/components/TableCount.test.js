import React from 'react';
import { Provider } from "react-redux";
import { render, getByText } from '@testing-library/react';
import { createTestStore } from '../store/testStore';
import TableCount from '../../components/table/TableCount';

let store;
let dispatch;
let component;
let container;
let asFragment;

describe("TableCount", () => {

  beforeEach(() => {
    store = createTestStore({
      mockStore: true,
      initialState: {
        gameReducer: {
          games: [
            { id: "1", title: "Game 1" },
            { id: "2", title: "Game 2" }
          ]
        }
      }
    });
    dispatch = store.dispatch;
    component = render(
      <Provider store={store}>
        <TableCount pageIndex={0} itemsPerPage={5} />
      </Provider>
    );
    container = component.container;
    asFragment = component.asFragment;
  });

  it("exists", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("has correct text", () => {
    expect(getByText(container, "Showing 1 to 2 of 2")).toBeVisible();
  });

});