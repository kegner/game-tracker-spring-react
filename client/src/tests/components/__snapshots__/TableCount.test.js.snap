// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCount exists 1`] = `
<DocumentFragment>
  Showing 1 to 2 of 2
</DocumentFragment>
`;
